universal-application-tool-0.0.1/app/services/applicant/question/ApplicantQuestion.java

```diff
@@ -76,6 +76,29 @@ public boolean isOptional() {
     return programQuestionDefinition.optional();
   }
 
+  /**
+   * Return true if this question is answered, or left unanswered in the program specified.
+   * Questions can only be left unanswered if they are optional.
+   *
+   * <p>For every applicant question, there are three possible states:
+   *
+   * <ol>
+   *   <li>unvisited - the applicant has not seen this question yet
+   *   <li>unanswered - the applicant has visited the question but chose to leave it unanswered.
+   *       Only optional questions can be left unanswered
+   *   <li>answered - the applicant has visited and provided an answer to the question
+   * </ol>
+   *
+   * For unanswered optional questions, we care in which program it was left unanswered. For the
+   * program which the optional question was left unanswered, it counts as "completed" for that
+   * program.
+   *
+   * @return true if this question is answered, or left unanswered in the program specified.
+   */
+  public boolean isAnsweredOrLeftUnansweredInProgram() {
```

Comment:
(nit) I find the term "unanswered" a bit confusing. It's explained well in the comments, but I wonder if the code would be more understandable if we renamed to something clearer (e.g., `isAnsweredOrSkippedOptionalInProgram`)

Reply:
Refactored to use this terminology instead